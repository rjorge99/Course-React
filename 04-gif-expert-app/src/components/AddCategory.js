import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = e => {
        setInputValue(e.target.value);
    };

    const handleSubmit = e => {
        e.preventDefault();

        console.log(inputValue);
        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input
                type='text'
                value={inputValue}
                onChange={e => handleInputChange(e)}
            />
        </form>
    );
};

export default AddCategory;

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
