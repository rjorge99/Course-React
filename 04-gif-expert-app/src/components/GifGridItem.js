import React from 'react'

export const GifGridItem = ({ title, url }) => {
    
    return (
        <div>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
