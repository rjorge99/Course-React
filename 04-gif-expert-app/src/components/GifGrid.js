import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs(category);

    return (
        <div>
            <h3>{category}</h3>
            {loading && <p>Loading...</p>}
            {images.map(img => (
                <GifGridItem key={img.id} {...img} />
            ))}
        </div>
    );
};
