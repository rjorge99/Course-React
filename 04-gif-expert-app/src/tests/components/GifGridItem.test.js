import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Prubas de GifGridExpert', () => {
    const title = 'Un titulo';
    const url = 'http://localhost/algo.jpg';

    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un parrafo con el parrafo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    test('Debe tener una imagen con los props', () => {
        const img = wrapper.find('img');

        // console.log(p.props());
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });
});
