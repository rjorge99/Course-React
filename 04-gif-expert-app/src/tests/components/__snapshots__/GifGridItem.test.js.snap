// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubas de GifGridExpert Debe mostrar el componente correctamente 1`] = `
<div>
  <img
    alt="Un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
